src/common/cmdparse.cc

```diff
@@ -159,11 +160,24 @@ dump_cmd_to_json(Formatter *f, uint64_t features, const string& cmd)
       }
     }
 
+    if (!allow_positional && desckv.count("positional") == 0) {
+      desckv["positional"] = "false";
+    }
+    bool was_n = false;
```

Comment:
could just remove `was_n`. and set `allow_positional` if key == "n" and value == "N", something like
```c++
for (auto [key, value] : desckv) {
  if (allow_positional &&
      (key == "n" && value == "N")) {
    // no more positional args after this!
    allow_positional = false;
  }
  f->dump_string(string(key).c_str(), value);
}
```

in which, 3 changes are included:

1. use "&&" not "and". we don't use alternative operators in ceph C++ code.
1. do not check "n=N" unless we need to do.
1. do not use "was_n" in the loop for tracking "n=N".  `allow_positional` would suffice.

Reply:
Is src/common/cmdparse.cc used in every command. What about the orchestrator? Wouldn't it be better to add or change the sigs in the pybind validate method (or somewhere around there) in ceph_argparse.py? Which the python ceph client (ceph.in) uses on every command (I think?).

Reply:
yeah, i think we can make this inference in ceph_argparse.py (e.g., on the client side) instead.

Reply:
i take it back.. easier to do it server side, i think, and then omit sending the positional property to pre-octopus clients